### PR TITLE
cleanup: move terminal summary code to terminal plugin

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/pytest-dev/pytest/issues/5067.

TODO:

- [x] changelog

### CODE REVIEW

**Comment by nicoddemus** [comment]
Looks OK so far. What do you think still needs improvement, besides a CHANGELOG?

File: src/_pytest/terminal.py [hunk 1]
```diff
@@ -873,6 +875,106 @@ def summary_stats(self):
...
+
+def _folded_skips(skipped):
+    d = {}
+    for event in skipped:
+        key = event.longrepr
+        assert len(key) == 3, (event, key)
+        keywords = getattr(event, "keywords", {})
+        # folding reports with global pytestmark variable
+        # this is workaround, because for now we cannot identify the scope of a skip marker
+        # TODO: revisit after marks scope would be fixed
```
Comment (nicoddemus) [inline src/_pytest/terminal.py]: @RonnyPfannschmidt is now the time to revisit this?

**Comment by blueyed** [comment]
I don't think it needs a changelog - it is only internal cleanup.

**Comment by nicoddemus** [comment]
> I don't think it needs a changelog - it is only internal cleanup.

I understand, but I think it still worth adding one, I can imagine someone accessing the moved functions in a custom plugin or something like that.

**Comment by nicoddemus** [comment]
(A trivial changelog entry that is)

File: changelog/5069.trivial.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+The code for the short test summary in the terminal was moved to the terminal plugin.
```
Comment (nicoddemus) [inline changelog/5069.trivial.rst]: Thanks 🙇

Comment (blueyed) [reply changelog/5069.trivial.rst]: You're welcome - sounds really trivial after all indeed.. ;)

/me needs to appreciate changelogs more / not getting annoyed by having to add them.. at least `[ci skip]` allows to skip Travis.

**Pull request merged**